crates/router/src/core/admin.rs

```diff
@@ -1888,8 +1897,9 @@ pub async fn create_business_profile(
     }
 
     Ok(service_api::ApplicationResponse::Json(
-        api_models::admin::BusinessProfileResponse::foreign_try_from(business_profile)
-            .change_context(errors::ApiErrorResponse::InternalServerError)?,
+        admin::business_profile_response(business_profile, &key_store)
+            .change_context(errors::ApiErrorResponse::InternalServerError)
```

Comment:
Can you add a context message with `attach_printable` or `attach_printable_lazy`?